File: examples/internal/escalation/dotnet/EscalationAgent.cs
using System.Text.Json.Nodes;
using Axme.Sdk;
using AxmeExamples;

namespace AxmeExamples.Internal;

public class EscalationAgent
{
    public static async Task Main(string[] args)
    {
        var baseUrl = SseHelper.Env("AXME_BASE_URL", "https://api.cloud.axme.ai");
        var apiKey = SseHelper.Env("AXME_API_KEY");
        var addr = SseHelper.Env("AXME_AGENT_ADDRESS", "incident-classifier-agent");
        if (string.IsNullOrEmpty(apiKey)) { Console.Error.WriteLine("AXME_API_KEY is required"); return; }

        var client = new AxmeClient(new AxmeClientConfig { ApiKey = apiKey, BaseUrl = baseUrl });
        Console.WriteLine($"incident-classifier starting  address={addr}");
        var sla = new Dictionary<string, int> { ["P1"] = 5, ["P2"] = 15, ["P3"] = 60, ["P4"] = 240 };
        string[] critical = ["api-gateway", "auth-service", "payments-processor"];
        var since = 0;

        while (true)
        {
            try
            {
                foreach (var d in await SseHelper.PollAgentStreamAsync(baseUrl, apiKey, addr, since))
                {
                    var id = SseHelper.Str(d, "intent_id");
                    if (d["seq"]?.GetValue<int>() is int seq) since = Math.Max(since, seq);
                    if (string.IsNullOrEmpty(id)) continue;
                    var intent = SseHelper.UnwrapIntent(await client.GetIntentAsync(id));
                    if (!SseHelper.IsActionable(intent)) continue;
                    var p = SseHelper.EffectivePayload(intent);

                    var sev = SseHelper.Str(p, "severity", "P3"); var svc = SseHelper.Str(p, "service");
                    var symptom = SseHelper.Str(p, "symptom");
                    var isCrit = critical.Contains(svc); var minutes = sla.GetValueOrDefault(sev, 60);
                    var affected = new List<string>(); if (!string.IsNullOrEmpty(svc)) affected.Add(svc); if (isCrit) affected.Add("downstream-dependents");
                    var esc = (sev is "P1" or "P2") && isCrit ? "on-call → team-lead → engineering-director" : sev is "P1" or "P2" ? "on-call → team-lead" : "on-call";

                    var res = new JsonObject { ["action"] = "complete", ["incident_id"] = SseHelper.Str(p, "incident_id", "?"),
                        ["severity"] = sev, ["is_critical_service"] = isCrit, ["sla_minutes"] = minutes,
                        ["affected_components"] = new JsonArray(affected.Select(a => (JsonNode)JsonValue.Create(a)!).ToArray()),
                        ["escalation_path"] = esc, ["has_error_rate_signal"] = symptom.Contains("error rate", StringComparison.OrdinalIgnoreCase) || symptom.Contains("5xx") };
                    await client.ResumeIntentAsync(id, res, new RequestOptions { OwnerAgent = addr });
                    Console.WriteLine($"resumed {id}");
                }
            }
            catch { await Task.Delay(2000); }
        }
    }
}

File: examples/internal/escalation/java/EscalationAgent.java
package ai.axme.examples.internal;

import ai.axme.examples.SseHelper;
import dev.axme.sdk.*;
import java.util.*;

/** Incident Classifier Agent — internal/escalation example (Java). */
public class EscalationAgent {
    static final String BASE = SseHelper.env("AXME_BASE_URL", "https://api.cloud.axme.ai");
    static final String KEY = System.getenv("AXME_API_KEY");
    static final String ADDR = SseHelper.env("AXME_AGENT_ADDRESS", "incident-classifier-agent");
    static final Map<String, Integer> SLA = Map.of("P1", 5, "P2", 15, "P3", 60, "P4", 240);
    static final Set<String> CRITICAL = Set.of("api-gateway", "auth-service", "payments-processor");

    public static void main(String[] args) throws Exception {
        if (KEY == null || KEY.isEmpty()) { System.err.println("AXME_API_KEY is required"); System.exit(1); }
        AxmeClient client = new AxmeClient(new AxmeClientConfig(BASE, KEY));
        System.out.printf("incident-classifier starting  address=%s%n", ADDR);
        int since = 0;
        while (true) {
            try {
                for (Map<String, Object> d : SseHelper.pollAgentStream(BASE, KEY, ADDR, since, 15)) {
                    String id = SseHelper.str(d, "intent_id"); Number seq = (Number) d.get("seq");
                    if (seq != null) since = Math.max(since, seq.intValue()); if (id.isEmpty()) continue;
                    Map<String, Object> intent = SseHelper.unwrapIntent(client.getIntent(id, RequestOptions.none()));
                    if (!SseHelper.isActionable(intent)) continue;
                    Map<String, Object> p = SseHelper.effectivePayload(intent);

                    String incId = SseHelper.str(p, "incident_id", "?"), sev = SseHelper.str(p, "severity", "P3");
                    String svc = SseHelper.str(p, "service"), symptom = SseHelper.str(p, "symptom");
                    int sla = SLA.getOrDefault(sev, 60); boolean crit = CRITICAL.contains(svc);
                    List<String> affected = new ArrayList<>(); if (!svc.isEmpty()) affected.add(svc); if (crit) affected.add("downstream-dependents");
                    String esc = ("P1".equals(sev) || "P2".equals(sev)) && crit ? "on-call → team-lead → engineering-director" :
                            ("P1".equals(sev) || "P2".equals(sev)) ? "on-call → team-lead" : "on-call";

                    Map<String, Object> res = new HashMap<>();
                    res.put("action", "complete"); res.put("incident_id", incId); res.put("severity", sev);
                    res.put("is_critical_service", crit); res.put("affected_components", affected);
                    res.put("sla_minutes", sla); res.put("escalation_path", esc);
                    res.put("has_error_rate_signal", symptom.toLowerCase().contains("error rate") || symptom.contains("5xx"));
                    client.resumeIntent(id, res, SseHelper.withOwner(ADDR));
                    System.out.printf("resumed %s%n", id);
                }
            } catch (Exception e) { Thread.sleep(2000); }
        }
    }
}
